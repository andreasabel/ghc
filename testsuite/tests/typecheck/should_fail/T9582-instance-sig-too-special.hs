-- 2014-09-18 Andreas Abel
-- Failing test case for ticket 9582

{-# LANGUAGE InstanceSigs, TypeFamilies #-}

module Fail where

class C a where
  type T a
  m :: T a

instance C [a] where
  type T [a] = [a]

  -- The following type signature for m should be invalid.
  -- We asked for something of a more general type.
  m :: String
  m = "bla"
