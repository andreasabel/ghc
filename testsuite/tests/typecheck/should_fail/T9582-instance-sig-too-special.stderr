
T9582-instance-sig-too-special.hs:17:8:
    Couldn't match type ‘a’ with ‘Char’
    Expected type: T [a]
      Actual type: String
    In the instance declaration for ‘C [a]’
